woocommerce/grow
# Upgrade the root dependencies and setup to use Node.js v20

### Changes proposed in this Pull Request:

Part of #108

This PR:

- Bump dependencies and regenerate package-lock.json for this repo with Node.js v20.
- Set minimum support for Node.js to `">=18"`, which align with the dependencies' minimum (`@wordpress/scripts@27.7.0`).
- Set .nvmrc to use Node.js v20.
- Fix eslint errors for the samples files.
- Update the "Publish Compat Checker package" workflow to use Node.js v20.

💡 These workflows depend on the release version, `woocommerce/grow/<action_name>@actions-v1`, so they will be upgraded after the `@actions-v2` is published.

- .github/workflows/js-linting.yml
- .github/workflows/php-coding-standards.yml
- .github/workflows/run-qit-all.yml

### Detailed test instructions:

1. JS linter
   - Use Node.js v20
   - `npm install`
   - `npm run lint:js` to see if the linter can be run and if the linting can pass
      - There may be some `import/no-unresolved` warnings. This kind of warnings can be ignore because the packages of sub-packages may not be installed.
   - Repeat this test with Node.js v18
1. Inspect the "Publish Compat Checker package" workflow
   - The current `s0/git-publish-subdir-action@develop` [revision](https://github.com/s0/git-publish-subdir-action/blob/5bc6742efb946f4cba68c7a9067a31ea5631071d/action.yml#L6) already uses Node.js v20
   - [The last workflow run](https://github.com/woocommerce/grow/actions/runs/8750523643) shows the following warning. The warning should be fixed after c8606e5865fac626cf646a0ed1d4adc4e53733ca
      ![image](https://github.com/woocommerce/grow/assets/17420811/7bd44078-c605-4309-9e2e-c94e0dbc5005)
